duckduckgo/iOS
# Add toggling of NetP Notifications to iOS

Task/Issue URL: https://app.asana.com/0/0/1205768725084797/f
BSK: https://github.com/duckduckgo/BrowserServicesKit/pull/541

**Description**:

This adds the implementation of the VPN Notifications screen. It serves as a way to toggle NetP notifications on the iOS Network Extension and also to help prompt enabling notifications app-wide.

This is quite a design-heavy project, but there’s also use of an app-group shared UserDefaults store to share the toggle state between the status view, app side and the notifications presenter, extension side.

**Steps to test this PR**:
1. Delete the iOS app and reinstall (to reset notifications settings)
2. Launch and enable internal testing (Settings -> Debug -> Internal User)
3. Go to the Network Protection screen (Settings -> Network Protection)
4. Select VPN Notifications
5. Check it matches[ these designs](https://app.asana.com/0/0/1205739632361534/f)
6. Press Turn on Notifications here **and disallow them**
7. Press the button again which will take you to the iOS DuckDuckGo notifications settings. Enabled them.
8. Go back to DuckDuckGo to the VPN Notifications screen and check they match these (other) [designs ](https://www.figma.com/file/d7d8ZGkWD0W5ExhBI0tESv/O-N---VPN-Settings-Templates?type=design&node-id=564-19005&mode=design&t=xGigSHUt6PJkyaR9-0)
9. Go back to the NetP Status view. Enabled NetP. **Observe the notification**
10. Go to VPN Notifications one more time.
11. Go back to the NetP Status view. Disable NeP and reenable it. **Observe there is no notification.**
12. Extra: you can try the same with the test notifications from the Debug menu. They should also be affected by this toggle.

**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
